app/models/item.rb

```diff
@@ -1,3 +1,9 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items
+  has_many :customers, through: :customer_items
+
+  def total_purchased
+    CustomerItem.where(item_id: self.id).count
```

Comment:
Nice job with this query. If you look at line 4, an `item` has_many `customers`. With this relationship, you could start right at `customers` and then count them:
```ruby
def total_purchased
  customers.distinct.count
end
```